### PR TITLE
Plugins: Cleanup up dependencies on core headers - background_fetch

### PR DESCRIPTION
**Pull request by SolidWallOfCode**

### CODE REVIEW

File: plugins/background_fetch/rules.cc [hunk 1]
```diff
@@ -24,6 +24,7 @@
 
 #include <cstdlib>
 #include <string_view>
+#include <string.h>
```
Comment (shinrich) [inline plugins/background_fetch/rules.cc]: Why are we adding the header file?

File: plugins/background_fetch/configs.cc [hunk 2]
```diff
@@ -23,6 +23,8 @@
 */
 
 #include "configs.h"
+#include <cstdio>
```
Comment (shinrich) [inline plugins/background_fetch/configs.cc]: Why are we adding these header files?

File: plugins/background_fetch/background_fetch.cc [hunk 3]
```diff
@@ -29,6 +29,7 @@
 
 #include <string>
 #include <unordered_map>
+#include <cinttypes>
```
Comment (shinrich) [inline plugins/background_fetch/background_fetch.cc]: Why are we adding the header file?

Comment (SolidWallOfCode) [reply plugins/background_fetch/background_fetch.cc]: Sized int types, such as `int64_t`.

**Comment by zwoop** [comment]
Cherry-picked to 8.0.x

**Pull request merged**